cli/src/commands/split.rs

```diff
@@ -51,6 +51,28 @@ pub(crate) struct SplitArgs {
     /// The revision to split
     #[arg(long, short, default_value = "@")]
     revision: RevisionArg,
+    /// The revision to copy as the first part of the split
+    ///
+    /// With this option, the first part of the split will contain the changes
+    /// between the parent of the REVISION and the FROM revision. The second
+    /// part of the split will contain the changes between the FROM revision and
+    /// the REVISION revision.
+    ///
+    /// This is especially useful if the FROM revision is a past version of
+    /// REVISION, with its commit id obtained via `jj obslog` or `jj log
+    /// --at-operation`.
+    //
+    // TODO(ilyagr): We could allow `--interactive --from`. It's unclear how
+    // useful that would be. It would mostly require writing tests and
+    // JJ-INSTRUCTIONS. More ambitiously, we could have a 3-pane interactive view
+    // with the FROM commit in the middle and the REVISION commit on the RHS.
```

Comment:
Maybe this point suggests that `split --restore-from` should be a separate command? It's also unclear to me how path arguments will be handled.

If we don't have `unsquash`, I might call this `unsquash` and the first commit would be restored including commit metadata.

Reply:
I'm currently mulling this over in my head and trying to get some experience of how I use this command in practice and maybe some fresh thoughts.

I'm not sure about much at this point, but here are my current thoughts.

> It's also unclear to me how path arguments will be handled.

To me, this seemed quite clear and is the main reason why I thought this fits into the `split` command: Doing the same thing as the `split` command already does seems to make a lot of sense. 

In the situation form the FAQ answer that I edited in this PR, if you want to restore only *some* of the files from the old version of the commit, you list them in the command line and the command should do the right thing.

For `jj split --restore-from --interactive`, I initially thought things are just as simple (it would let you edit the diff for the FROM commit, make the result of that the intermediate commit, and -- like always with `split` -- make the child commit identical to the -r commit), but there's a question of what to put as the base of the diff if `--from` commit has a different parent than `-r` commit.

> If we don't have unsquash, I might call this unsquash

This is an interesting idea, thanks! This is indeed the operation you'd use to undo a `squash`.

I'm not sure about it, though. In the situation from the FAQ, `unsquash` sounds less natural to me. Also, it depends a little on whether we allow path arguments and what we do with them.

> and the first commit would be restored including commit metadata.

I was wondering about this, but there are a couple of problems. Preserving the metadata would very often cause divergence in the situations where this command would be, I think, most useful. Also, this would be a bit weird if the `--from` commit had a different parent than the `-r` commit.

Reply:
By the way, a way to fix the issue of "this would be a bit weird if the --from commit had a different parent than the -r commit" with your "preserve metadata" idea would be to rebase the `-r` commit onto the `--from` commit. At this point, the command becomes nearly identical to `rebase --verbatim` that @martinvonz has been mentioning, #1027.

My current opinion is that both `rebase --verbatim` and `split --from` are useful (perhaps under different names, these are just the ones I remember the ideas by), but if we could do everything with one of them, that's fine too.

Reply:
Perhaps, I find `split --restore-from` unintuitive because I see `split` is the command to divide the specified commit into two, so the first and the second commits are equally important. If I consider it as a command to insert intermediate state, it makes sense that `--restore-from` "restores" tree into the first (= newly-created) commit.

Reply:
I'll call this "resolved", but of course we can continue the conversation.

Reply:
To me, `split --restore-from` is unintuitive because it inserts new state (which doesn't exist in the commit to be split.) That said, I'm not strongly against it because the added functionality is useful.